pkg/registry/minion/rest.go

```diff
@@ -91,7 +91,18 @@ func (rs *REST) Get(ctx api.Context, id string) (runtime.Object, error) {
 
 // List satisfies the RESTStorage interface.
 func (rs *REST) List(ctx api.Context, label, field labels.Selector) (runtime.Object, error) {
-	return rs.registry.ListMinions(ctx)
+	nodes, err := rs.registry.ListMinions(ctx)
```

Comment:
I see that for Watch, we pass the label selector to the registry, which in turn passes a filter function to etcd (https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/registry/etcd/etcd.go#L404) which does the actual filtering.
We should be using the same pattern at both places.